### ISSUE TITLE
Interface types are documented as "Object"

### ISSUE THREAD

**Issue by lbndev** [stoicflame/enunciate#561]
Working on #557 I noticed that when a method returns a data type that is an interface, it is simply shown as Object. Interface types should be documented the same way as abstract parent types.

**Comment by stoicflame** [comment]
Interfaces aren't data types, so Enunciate doesn't document them as such.

Maybe another way to say it is that there is no such thing as an "interface" in a Web service API. It's a construct specific to the language that doesn't translate to HTTP, SOAP, REST, XML, JSON, etc. So since Enunciate is documenting the Web service API, it's limited to the vocabulary of Web services and has no way of documenting an "interface."

**Comment by stoicflame** [comment]
See also #551.

**Comment by lbndev** [comment]
Sure, but Java allows to define an interface type and use it as parameter or return type for services that Enunciate documents. So when I'm doing this :
```java
public interface ISomething { }
public class SomethingImpl implements ISomething { }
public class AnotherImpl implements ISomething { }
@Path("/service")
public interface IService {
    @POST
    @Path("/execute")
    public ISomething doWork(ISomething input);
}
```
Enunciate shows parameter and return type of /service/execute as "Object" (or Array of Object if using a collection of ISomething). @typeHint or @returnWrapped won't help : any implementation can be a valid value.
It is not always possible to change the interface to an abstract base class, so how can we document such services ?
My suggestion is to treat interface types as if they were abstract superclasses (dedicated page + as I added in my PR #562 : list of implementing types + warning message that this is "abstract").
Do you have another idea ?

**Comment by stoicflame** [comment]
I started looking into this.

It's a reasonable idea, but it's quite a bit of work, so I'm pulling it off the release plan for now.

If anybody is willing to sponsor the work, I'd be happy to pick it up again.

**Comment by stoicflame** [comment]
Thanks to the support of a generous sponsor, this issue has been implemented and will be included in 2.9.

**Comment by stoicflame** [comment]
[Enunciate 2.9.0 has been released.](https://github.com/stoicflame/enunciate/releases/tag/v2.9.0)
